client/src/status/StatusHandler.ts

```diff
@@ -0,0 +1,33 @@
+import { StatusNotification } from 'dbt-language-server-common';
+import { LanguageStatusItems } from './LanguageStatusItems';
+import { ProjectStatus } from './ProjectStatus';
+
+export class StatusHandler {
+  private projectStatuses: Map<string, ProjectStatus> = new Map();
+  private statusItems?: LanguageStatusItems;
+
+  updateLanguageItems(projectPath: string): void {
+    this.getProjectStatus(projectPath).updateStatusUi();
+  }
+
+  onStatusChanged(statusNotification: StatusNotification): void {
+    this.getProjectStatus(statusNotification.projectPath).updateStatusData(statusNotification);
+    this.updateLanguageItems(statusNotification.projectPath);
+  }
+
+  private getStatusItems(): LanguageStatusItems {
+    if (this.statusItems === undefined) {
+      this.statusItems = new LanguageStatusItems();
```

Comment:
It seems that if `LanguageClient` is created we always need `statusItems` to be created.

Reply:
Yep, refactored a bit.